src/platforms/hosted/dap_swd.c

```diff
@@ -119,59 +128,94 @@ static void dap_swd_seq_out(const uint32_t tms_states, const size_t clock_cycles
 		DEBUG_ERROR("dap_swd_seq_out failed\n");
 }
 
+/*Fallback when SWD sequence to implemented*/
+static void dap_swj_seq_out(const uint32_t tms_states, const size_t clock_cycles)
+{
+	const uint8_t data[4] = {(tms_states >> 0) & 0xff, (tms_states >> 8) & 0xff,
+		(tms_states >> 16) & 0xff, (tms_states >> 24) & 0xff};
+	if (!perform_dap_swj_sequence(clock_cycles, data))
+		DEBUG_ERROR("dap_swj_seq_out failed\n");
+}
+
 static void dap_swd_seq_out_parity(const uint32_t tms_states, const size_t clock_cycles)
 {
-	/* Setup the sequence */
-	dap_swd_sequence_s sequence = {
-		clock_cycles + 1,
-		DAP_SWD_OUT_SEQUENCE,
+	dap_swd_sequence_s sequences[2] = {
+		/* Turnaround always needed after dap_swd_seq_in */
+		{
+			1,
+			DAP_SWD_OUT_SEQUENCE,
+			{0},
+		},
+		/* Data phase*/
+		{
+			clock_cycles + 1,
+			DAP_SWD_OUT_SEQUENCE,
+		},
 	};
-	write_le4(sequence.data, 0, tms_states);
-	sequence.data[4] = __builtin_parity(tms_states);
+	write_le4(sequences[1].data, 0, tms_states);
+	sequences[1].data[4] = __builtin_parity(tms_states);
 	/* And perform it */
-	if (!perform_dap_swd_sequences(&sequence, 1U))
+	swdio_status = SWDIO_STATUS_DRIVE;
+	if (!perform_dap_swd_sequences(sequences, 2U))
 		DEBUG_ERROR("dap_swd_seq_out_parity failed\n");
 }
 
 static uint32_t dap_swd_seq_in(const size_t clock_cycles)
 {
 	/* Setup the sequence */
-	dap_swd_sequence_s sequence = {
-		clock_cycles,
-		DAP_SWD_IN_SEQUENCE,
+	dap_swd_sequence_s sequences[2] = {
+		/* one cycle turn-around*/
+		{
+			1,
+			DAP_SWD_IN_SEQUENCE
+		},
+		/* Ack read phase */
+		{
+			clock_cycles,
+			DAP_SWD_IN_SEQUENCE
+		},
 	};
+	swdio_status = SWDIO_STATUS_FLOAT;
 	/* And perform it */
-	if (!perform_dap_swd_sequences(&sequence, 1U)) {
+	if (!perform_dap_swd_sequences(sequences, 2U)) {
 		DEBUG_ERROR("dap_swd_seq_in failed\n");
 		return 0U;
 	}
 
 	uint32_t result = 0;
 	for (size_t offset = 0; offset < clock_cycles; offset += 8U)
-		result |= sequence.data[offset >> 3U] << offset;
+		result |= sequences[1].data[offset >> 3U] << offset;
 	return result;
 }
 
 static bool dap_swd_seq_in_parity(uint32_t *const result, const size_t clock_cycles)
 {
 	/* Setup the sequence */
-	dap_swd_sequence_s sequence = {
-		clock_cycles + 1U,
-		DAP_SWD_IN_SEQUENCE,
+	dap_swd_sequence_s sequences[2] = {
+		{
+			clock_cycles + 1U,
+			DAP_SWD_IN_SEQUENCE,
+		},
+		{ /* 8 idle cycle to return to SWDIO_STATUS_DRIVE */
+			8,
+			DAP_SWD_OUT_SEQUENCE,
+			{0},
+		},
 	};
+	swdio_status = SWDIO_STATUS_DRIVE;
 	/* And perform it */
-	if (!perform_dap_swd_sequences(&sequence, 1U)) {
+	if (!perform_dap_swd_sequences(sequences, 2U)) {
 		DEBUG_ERROR("dap_swd_seq_in_parity failed\n");
-		return false;
+		return true;
```

Comment:
This should return false for the parity check failing. Please don't invert the logic of the returns on this function. It doesn't work this way.